tlsobs-api/handlers.go

```diff
@@ -430,6 +431,71 @@ func PathsHandler(w http.ResponseWriter, r *http.Request) {
 	return
 }
 
+// TruststoreHandler handles the /truststore endpoint of the api.
+// It queries the database for all certificates trusted by a certain program.
+// It takes the following parameters as HTTP query parameters:
+//     store: one of {"mozilla", "android", "apple", "microsoft", "ubuntu"}
+//     format: one of {"json", "pem"}
+func TruststoreHandler(w http.ResponseWriter, r *http.Request) {
+	setResponseHeader(w)
+	log := logger.GetLogger()
+	log.WithFields(logrus.Fields{
+		"form values": r.Form.Encode(),
+		"headers":     r.Header,
+		"uri":         r.URL.RequestURI(),
+	}).Debug("Truststore Endpoint received request")
+
+	val := r.Context().Value(dbKey)
+	if val == nil {
+		httpError(w, http.StatusInternalServerError, "Could not find database handler in request context")
+		return
+	}
+	db := val.(*pg.DB)
+	certs, err := db.GetAllCertsInStore(r.FormValue("store"))
+	if err == pg.ErrInvalidCertStore {
+		httpError(w, http.StatusBadRequest, fmt.Sprintf("Invalid certificate trust store provided: %s", r.FormValue("store")))
+		return
+	} else if err != nil {
+		logrus.Error("Error querying truststore:", err)
+		httpError(w, http.StatusBadRequest, "Error querying trust store")
+		return
+	}
+	switch r.FormValue("format") {
+	case "json":
+		certsJSON, err := json.Marshal(certs)
+		if err != nil {
+			httpError(w, http.StatusInternalServerError, "Could not marshal certificates")
+			return
+		}
+		w.WriteHeader(http.StatusOK)
+		w.Header().Set("Content-Type", "application/json")
+		w.Write(certsJSON)
+		logRequest(r, http.StatusOK, len(certsJSON))
+	case "pem":
+		var buffer bytes.Buffer
+		for _, cert := range certs {
+			x509, err := cert.ToX509()
```

Comment:
This is the output the latest commit produces:

```
# Certificate "GlobalSign Root CA"
# Issuer: CN=GlobalSign Root CA,OU=[Root CA],O=[GlobalSign nv-sa],C=[BE]
# Serial Number: 40000000001154b5ac394
# Subject: CN=GlobalSign Root CA,OU=[Root CA],O=[GlobalSign nv-sa],C=[BE]
# Not Valid Before: 1998-09-01 12:00:00 +0000 UTC
# Not Valid After : 2028-01-28 12:00:00 +0000 UTC
# Fingerprint (SHA256): ebd41040e4bb3ec742c9e381d31ef2a41a48b6685c96e7cef3c1df6cd4331c99
```

Apparently you can have more than one OU, O or C, so it's printed in a slice format.